core/native_exec.c

```diff
@@ -212,15 +213,16 @@ entering_native(dcontext_t *dcontext)
      * We can't revert memory prots, since other threads are under DR
      * control, but we do handle our-fault write faults in native threads.
      */
-    set_asynch_interception(dcontext->owning_thread, false);
-#endif
-    /* FIXME: setting same var that set_asynch_interception is! */
+    /* FIXME i#2375: for -native_exec_opt on UNIX we need to update the gencode
+     * to do what os_thread_{,not_}under_dynamo() and os_thread_re_take_over() do.
+     */
+    if (IF_WINDOWS_ELSE(true, !DYNAMO_OPTION(native_exec_opt)))
+        os_thread_not_under_dynamo(dcontext);
```

Comment:
So does it means native_exec_opt not work?
Should we add similar comment at the place where we suppose to mirror what os_thread_{,not_}under_dynamo() and os_thread_re_take_over() do.